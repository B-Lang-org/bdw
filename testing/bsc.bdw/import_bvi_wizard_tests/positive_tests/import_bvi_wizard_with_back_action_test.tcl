global BSPEC

# Opens the specified project
handlers::open_project project.bspec 

# Opens the wizard 
create_import_bvi_wizard_window

# Loads the verilog file
$BSPEC(IMPORT_BVI) read_from_file mult4/info_verilog_for_mkMult2

# Go to Bluespec Module Definition step
$BSPEC(IMPORT_BVI) action_next

# Sets the module name
$BSPEC(IMPORT_BVI) set_bluespec_module_name "module_test" 

# Sets the interface name
$BSPEC(IMPORT_BVI) set_bluespec_interface_name "Ifc_test" 

# Sets the default clock
$BSPEC(IMPORT_BVI) set_bluespec_input_clock_default "CLK"

# Sets the default reset
$BSPEC(IMPORT_BVI) set_bluespec_input_reset_default "RST"

# Go to Method Port Bindings step
$BSPEC(IMPORT_BVI) action_next

# Create method according to existence port
$BSPEC(IMPORT_BVI) auto_create_from_verilog

# Go to Path step
$BSPEC(IMPORT_BVI) action_next

# Go to Scheduling Annotation step
$BSPEC(IMPORT_BVI) action_next

# Go to Finish step
$BSPEC(IMPORT_BVI) action_next

# Sets the file name
$BSPEC(IMPORT_BVI) set_save_file_location "import_bvi_wizard_test.bsv"

$BSPEC(IMPORT_BVI) action_back
$BSPEC(IMPORT_BVI) action_back
$BSPEC(IMPORT_BVI) action_back
$BSPEC(IMPORT_BVI) action_back
$BSPEC(IMPORT_BVI) action_back

$BSPEC(IMPORT_BVI) action_next
$BSPEC(IMPORT_BVI) action_next
$BSPEC(IMPORT_BVI) action_next
$BSPEC(IMPORT_BVI) action_next
$BSPEC(IMPORT_BVI) action_next

# Finish the wizard action
$BSPEC(IMPORT_BVI) save_now

# Rename the wizard output file
exec cp import_bvi_wizard_test.bsv import_bvi1.bsv
exec mv import_bvi_wizard_test.bsv import_bvi_wizard_with_back_action_test.tcl.bdw-out

# Closes the wizard
$BSPEC(IMPORT_BVI) action_close

# Exits the workstation
exit

