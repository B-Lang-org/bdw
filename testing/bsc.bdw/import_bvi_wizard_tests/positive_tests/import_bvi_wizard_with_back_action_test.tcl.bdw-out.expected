// Bluespec wrapper, created by Import BVI Wizard
// Created on: Fri Feb 27 16:43:51 AMT 2009
// Created by: employee
// Bluespec version: 2008.11.C 2008-11-24 15480


interface Ifc_test;
	(*always_ready*)
	method Action istart_m1 (Bit#(16) start_m1);
	(*always_ready*)
	method Action istart_m2 (Bit#(16) start_m2);
	method Action iEN_start ();
	method Action iEN_result ();
	method Bool oRDY_start ();
	method Bit#(32) oresult ();
	method Bool oRDY_result ();
endinterface

import "BVI" mkMult2 =
module module_test  (Ifc_test);

	default_clock clk_CLK;
	default_reset rst_RST;

	input_clock clk_CLK (CLK)  <- exposeCurrentClock;
	input_reset rst_RST (RST) clocked_by(clk_CLK)  <- exposeCurrentReset;


	method istart_m1 (start_m1 /*15:0*/)
		 enable((*inhigh*)istart_m1_enable) clocked_by(clk_CLK) reset_by(rst_RST);
	method istart_m2 (start_m2 /*15:0*/)
		 enable((*inhigh*)istart_m2_enable) clocked_by(clk_CLK) reset_by(rst_RST);
	method iEN_start ()
		 enable(EN_start) clocked_by(clk_CLK) reset_by(rst_RST);
	method iEN_result ()
		 enable(EN_result) clocked_by(clk_CLK) reset_by(rst_RST);
	method RDY_start oRDY_start ()
		 clocked_by(clk_CLK) reset_by(rst_RST);
	method result /* 31 : 0 */ oresult ()
		 clocked_by(clk_CLK) reset_by(rst_RST);
	method RDY_result oRDY_result ()
		 clocked_by(clk_CLK) reset_by(rst_RST);

	schedule istart_m1 C istart_m1;
	schedule istart_m1 CF istart_m2;
	schedule istart_m1 CF iEN_start;
	schedule istart_m1 CF iEN_result;
	schedule oRDY_start SB istart_m1;
	schedule oresult SB istart_m1;
	schedule oRDY_result SB istart_m1;
	schedule istart_m2 C istart_m2;
	schedule istart_m2 CF iEN_start;
	schedule istart_m2 CF iEN_result;
	schedule oRDY_start SB istart_m2;
	schedule oresult SB istart_m2;
	schedule oRDY_result SB istart_m2;
	schedule iEN_start C iEN_start;
	schedule iEN_start CF iEN_result;
	schedule oRDY_start SB iEN_start;
	schedule oresult SB iEN_start;
	schedule oRDY_result SB iEN_start;
	schedule iEN_result C iEN_result;
	schedule oRDY_start SB iEN_result;
	schedule oresult SB iEN_result;
	schedule oRDY_result SB iEN_result;
	schedule oRDY_start CF oRDY_start;
	schedule oRDY_start CF oresult;
	schedule oRDY_start CF oRDY_result;
	schedule oresult CF oresult;
	schedule oresult CF oRDY_result;
	schedule oRDY_result CF oRDY_result;
endmodule


